src/commands/sourcemaps/upload.ts

```diff
@@ -238,11 +238,20 @@ export class UploadCommand extends Command {
     metricsLogger: BufferedMetricsLogger,
     sourcemap: Payload
   ): Promise<UploadStatus> {
-    if (!fs.existsSync(sourcemap.minifiedFilePath)) {
-      this.context.stdout.write(
-        renderFailedUpload(sourcemap, `Missing corresponding JS file for sourcemap (${sourcemap.minifiedFilePath})`)
-      )
-      metricsLogger.increment('skipped_missing_js', 1)
+    try {
+      validatePayload(sourcemap)
+    } catch (error) {
+      if (error instanceof InvalidPayload) {
+        this.context.stdout.write(renderFailedUpload(sourcemap, error.message))
+        metricsLogger.increment(error.reason, 1)
+      } else {
+        this.context.stdout.write(
```

Comment:
Do we want to add something like `metricsLogger.increment("unknown", 1)` in the else case?

Reply:
Agreed, I pushed another commit with better metrics.